v6/v6_client.ml

```diff
@@ -0,0 +1,45 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open V6_interface
+open Xcp_client
+
+let retry_econnrefused f =
```

Comment:
Two minor thoughts: I would have called this `connect` and would have used a counter for the retry-loop. Right now you can only retry once but a counter and a limit would be more flexible. Not requesting a change, though.